privacy-on-beam/pbeam/mean_test.go

```diff
@@ -536,58 +533,77 @@ func TestMeanPerKeyNoNoiseFloatValues(t *testing.T) {
+
+		exactCount0 :=  7.0
+		exactMean0 := 14.0/exactCount0
+		exactCount := 250.0 
+		exactMean := (100.0 + 2.0*150.0) / exactCount
+		result := []testFloat64Metric{
+			{0, exactMean0},
+			{1, exactMean},
+		}
 
-	// We have ε=50, δ=0 and l0Sensitivity=1. No thresholding is done because partitions are specified.
-	// We have 2 partitions. So, to get an overall flakiness of 10⁻²³,
-	// we can have each partition fail with 10⁻²⁵ probability (k=25).
-	maxContributionsPerPartition := int64(1)
-	maxPartitionsContributed := int64(1)
-	epsilon := 100.0
-	delta := 0.0
-	lower := 1.0
-	upper := 3.0
+		p, s, col, want := ptest.CreateList2(triples, result)
+		col = beam.ParDo(s, extractIDFromTripleWithFloatValue, col)
 
-	// ε is not split because partitions are specified.
-	pcol := MakePrivate(s, col, NewPrivacySpec(epsilon, delta))
-	pcol = ParDo(s, tripleWithFloatValueToKV, pcol)
-	got := MeanPerKey(s, pcol, MeanParams{
-		MaxPartitionsContributed:     maxPartitionsContributed,
-		MaxContributionsPerPartition: maxContributionsPerPartition,
-		MinValue:                     lower,
-		MaxValue:                     upper,
-		NoiseKind:                    LaplaceNoise{},
-		partitionsCol: partitionsCol,
-	})
+		partitions := []int{0, 1}
+		partitionsCol := beam.CreateList(s, partitions)
+
+		// We have ε=50, δ=0 and l0Sensitivity=1. No thresholding is done because partitions are specified.
+		// We have 2 partitions. So, to get an overall flakiness of 10⁻²³,
+		// we can have each partition fail with 10⁻²⁵ probability (k=25).
+		maxContributionsPerPartition := int64(1)
+		maxPartitionsContributed := int64(1)
+		epsilon := 100.0
+		delta := 0.0
+		lower := tc.lower
+		upper := tc.upper
+
+		// ε is not split because partitions are specified.
+		pcol := MakePrivate(s, col, NewPrivacySpec(epsilon, delta))
+		pcol = ParDo(s, tripleWithFloatValueToKV, pcol)
+		got := MeanPerKey(s, pcol, MeanParams{
+			MaxPartitionsContributed:     maxPartitionsContributed,
+			MaxContributionsPerPartition: maxContributionsPerPartition,
+			MinValue:                     lower,
+			MaxValue:                     upper,
+			NoiseKind:                    LaplaceNoise{},
+			partitionsCol: partitionsCol,
+		})
 
-	want = beam.ParDo(s, float64MetricToKV, want)
-	tolerance, err := laplaceToleranceForMean(25, lower, upper, maxContributionsPerPartition, maxPartitionsContributed, epsilon, 5, exactCount, exactMean)
-	if err != nil {
-		t.Fatalf("laplaceToleranceForMean: got error %v", err)
-	}
-	if err := approxEqualsKVFloat64(s, got, want, tolerance); err != nil {
-		t.Fatalf("TestMeanPerKeyWithPartitionsNoNoise: %v", err)
-	}
-	if err := ptest.Run(p); err != nil {
-		t.Errorf("TestMeanPerKeyWithPartitionsNoNoise: MeanPerKey(%v) = %v, want %v, error %v", col, got, want, err)
+		want = beam.ParDo(s, float64MetricToKV, want)
+		tolerance, err := laplaceToleranceForMean(25, lower, upper, maxContributionsPerPartition, maxPartitionsContributed, epsilon, 5, exactCount, exactMean)
```

Comment:
Because laplaceToleranceForMean computes lInfSensitivity based on lower and upper, we don't need to worry about lInfSensitivity ourselves. That's why Mean tests were not flaky.